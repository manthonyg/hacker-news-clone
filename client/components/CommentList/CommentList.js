import React from "react";
import Comment from "../Comments/Comment";
import PropTypes from "prop-types";

function CommentList({ comments }) {
  return (
    <>
      {comments &&
        !!comments.length &&
        comments.map((comment, idx) => (
          <Comment key={`${comment?.by.slice(-5)}${idx}`} comment={comment} />
        ))}
    </>
  );
}

CommentList.propTypes = {
  comments: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      by: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      time: PropTypes.number.isRequired,
    })
  ),
};

export default CommentList;
