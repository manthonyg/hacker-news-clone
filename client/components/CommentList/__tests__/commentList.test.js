import React from "react";
import renderer from "react-test-renderer";
import CommentList from "../CommentList";
import RenderWithRouter from "../../../test_utils/renderWithRouter";

it("renders correctly with all props", () => {
  const requiredProps = {
    comments: [
      {
        id: 5,
        by: "moopie",
        text: "Jest: a tale of misery",
        time: 231238213,
      },
    ],
  };

  const tree = renderer
    .create(
      <>
        <RenderWithRouter>
          <CommentList {...requiredProps} />
        </RenderWithRouter>
      </>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
