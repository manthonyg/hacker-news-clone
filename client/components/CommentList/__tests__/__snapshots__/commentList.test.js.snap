// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="sc-AxmLO vFtN"
>
  <p
    className="sc-fzoLsD cuxcLk"
  >
    <h5
      className="sc-AxhUy iTChLH"
    >
      <a
        className="sc-fzozJi eraVuk"
        href="/user?id=moopie"
        onClick={[Function]}
      >
        moopie
      </a>
       at
       
      1977-04-30
      :
    </h5>
  </p>
   
  <p
    className="sc-fzpans jZRKiG"
  >
     
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "Jest: a tale of misery",
        }
      }
    />
  </p>
</div>
`;
